src/cosmos/engine/CosmosEngine.ts

```diff
@@ -795,95 +786,48 @@ export class CosmosEngine extends CurrencyEngine<
-
-    const { url, headers } = rpcWithApiKey(
-      this.networkInfo.defaultTransactionFeeUrl,
-      this.tools.initOptions
-    )
-    const heightQueryParam = blockheight == null ? '' : `?height=${blockheight}`
-    const res = await this.fetchCors(`${url}${heightQueryParam}`, {
-      method: 'GET',
-      headers
-    })
-    const raw = await res.json()
-    const clean = asThornodeNetwork(raw)
-    for (const msg of messages) {
-      switch (msg.typeUrl) {
-        case '/types.MsgDeposit':
-          networkFee = add(networkFee, clean.native_outbound_fee_rune)
-          break
-        case '/types.MsgSend':
-          networkFee = add(networkFee, clean.native_tx_fee_rune)
-      }
-    }
-    return networkFee
-  }
-
-  private async calculateFee(opts: {
+  async calculateFee(opts: {
     messages: EncodeObject[]
     memo?: string
     networkFeeOption?: EdgeSpendInfo['networkFeeOption']
   }): Promise<CosmosFee> {
-    const { messages, memo, networkFeeOption } = opts
-    let gasFeeCoin = coin('0', this.networkInfo.nativeDenom)
-    let gasLimit = '0'
-    let networkFee = '0'
-    if (this.networkInfo.defaultTransactionFeeUrl == null) {
-      const { queryClient } = this.getClients()
-      const { gasInfo } = await queryClient.tx.simulate(
-        messages,
-        memo,
-        encodeSecp256k1Pubkey(base16.parse(this.walletInfo.keys.publicKey)),
-        this.getSequence()
-      )
-      if (gasInfo?.gasUsed == null) {
-        throw new Error(`simulate didn't return gasUsed `)
-      }
-      // The simulate endpoint is imperfect and under-estimates. It's typical to use 1.5x the estimated amount
-      gasLimit = ceil(mul(gasInfo?.gasUsed.toString(), '1.5'), 0)
+    const { queryClient } = this.getClients()
+    const { gasInfo } = await queryClient.tx.simulate(
+      opts.messages,
+      opts.memo,
+      encodeSecp256k1Pubkey(base16.parse(this.walletInfo.keys.publicKey)),
+      this.getSequence()
+    )
+    if (gasInfo?.gasUsed == null) {
+      throw new Error(`simulate didn't return gasUsed `)
+    }
+    // The simulate endpoint is imperfect and under-estimates. It's typical to use 1.5x the estimated amount
+    const gasLimit = ceil(mul(gasInfo?.gasUsed.toString(), '1.5'), 0)
 
-      const { low, average, high } = getGasPriceStep(this.tools.chainData)
+    const { low, average, high } = getGasPriceStep(this.tools.chainData)
 
-      let gasPrice = average
-      switch (networkFeeOption) {
-        case 'low': {
-          gasPrice = low
-          break
-        }
-        case 'high': {
-          gasPrice = high
-          break
-        }
-        case 'custom': {
-          throw new Error('Custom fee not supported')
-        }
+    let gasPrice = average
+    switch (opts.networkFeeOption) {
+      case 'low': {
+        gasPrice = low
+        break
+      }
+      case 'high': {
+        gasPrice = high
+        break
+      }
+      case 'custom': {
+        throw new Error('Custom fee not supported')
```

Comment:
**You can't ignore this comment**

If only this was valid JS code:

```js
const gasPrice = match opts.networkFeeOption === {
  'low': low,
  'high': high,
  'custom': throw new Error('Custom fee not supported')
}
```

It's called pattern matching, but the syntax idea that I just thought of seems nice and elegant. But this is in TC39's hands.

```js
const gasPrice = match(opts.networkFeeOption)(
  (case = Type('low')) => low,
  (case = Type('high')) => high,
  (case = Type('custom')) => { throw new Error(...) }
)
```

^ This API is doable though ^

*whatever*